test.js

```diff
@@ -213,3 +229,49 @@ test.serial('if once option is true and silent is false, hook only prints the fi
 	t.deepEqual(hookReturnValue, false);
 	t.deepEqual(log, [['foo'], ['bar']]);
 });
+
+test('promise resolves when stdout & stderr are hooked and released via promise', async t => {
+	t.plan(3);
+	const promise = m(str => {
+		if (str === 'foo' || str === 'bar') {
+			t.pass();
+		}
+	});
+	process.stdout.write('foo');
+	process.stderr.write('bar');
+	promise.unhook();
+	await promise.then(types => t.deepEqual(types, ['stdout', 'stderr']));
+});
+
+test('promise resolves when stdout & stderr are hooked and released via callback', async t => {
+	t.plan(3);
+	const promise = m((str, enc, unhook) => {
+		if (str === 'foo' || str === 'bar') {
+			t.pass();
+		}
+		unhook();
+	});
+	process.stdout.write('foo');
+	process.stderr.write('bar');
+	await promise.then(types => t.deepEqual(types, ['stdout', 'stderr']));
+});
+
+test('promise resolves when stdout is released via promise', async t => {
+	t.plan(2);
+	const promise = m.stdout(str => {
+		t.is(str, 'foo');
+	});
+	process.stdout.write('foo');
+	promise.unhook();
+	await promise.then(type => t.is(type, 'stdout'));
+});
+
+test('promise resolves when stderr is released via promise', async t => {
+	t.plan(2);
+	const promise = m.stderr(str => {
+		t.is(str, 'foo');
+	});
+	process.stderr.write('foo');
+	promise.unhook();
+	await promise.then(type => t.is(type, 'stderr'));
```

Comment:
I don't really see the point in returning the `type`. I think it should just resolve with nothing.

Also to be able to, for example, return the collected output here.